repos/system_upgrade/el7toel8/actors/migratentp/libraries/ntp2chrony.py

```diff
@@ -0,0 +1,664 @@
+                    continue
+                assert False
+            else:
+                conf += "{} {}".format(source["type"], address)
+                for option in source["options"]:
+                    if option[0] in ["minpoll", "maxpoll", "version", "key",
+                                     "iburst", "noselect", "prefer", "xleave"]:
+                        conf += " {}".format(" ".join(option))
+                    elif option[0] == "burst":
+                        conf += " presend 6"
+                    elif option[0] == "true":
+                        conf += " trust"
+                    else:
+                        assert False
+                conf += "\n"
+        conf += "\n"
+
+        return conf
+
+    def get_chrony_conf_allows(self):
+        allowed_networks = filter(lambda n: "ignore" not in self.restrictions[n] and
+                                    "noserve" not in self.restrictions[n],
+                                  self.restrictions.keys())
+
+        conf = ""
+        for network in sorted(allowed_networks, key=lambda n: (n.version, n)):
+            if network.num_addresses > 1:
+                conf += "allow {}\n".format(network)
+            else:
+                conf += "allow {}\n".format(network.network_address)
+
+        if conf:
+            conf = "# Allow NTP client access.\n" + conf
+            conf += "\n"
+
+        return conf
+
+    def get_chrony_conf_cmdallows(self):
+        allowed_networks = filter(lambda n: "ignore" not in self.restrictions[n] and
+                                    "noquery" not in self.restrictions[n] and
+                                    n != ipaddress.ip_network(u"127.0.0.1/32") and
+                                    n != ipaddress.ip_network(u"::1/128"),
+                                  self.restrictions.keys())
+
+        ip_versions = set()
+        conf = ""
+        for network in sorted(allowed_networks, key=lambda n: (n.version, n)):
+            ip_versions.add(network.version)
+            if network.num_addresses > 1:
+                conf += "cmdallow {}\n".format(network)
+            else:
+                conf += "cmdallow {}\n".format(network.network_address)
+
+        if conf:
+            conf = "# Allow remote monitoring.\n" + conf
+            if 4 in ip_versions:
+                conf += "bindcmdaddress 0.0.0.0\n"
+            if 6 in ip_versions:
+                conf += "bindcmdaddress ::\n"
+            conf += "\n"
+
+        return conf
+
+    def get_chrony_conf(self, chrony_keys_path):
+        local_stratum = 0
+        maxdistance = 0.0
+        minsources = 1
+        orphan_stratum = 0
+        logs = []
+
+        for source in self.time_sources:
+            address = source["address"]
+            if address.startswith("127.127.1."):
+                if address in self.fudges and "stratum" in self.fudges[address]:
+                    local_stratum = int(self.fudges[address]["stratum"])
```

Comment:
Can this raise a ValueError during convertion?

Reply:
Yes, good point. It should be now fixed.